tests/sentry/queue/test_routers.py

```diff
@@ -0,0 +1,115 @@
+import itertools
+from collections.abc import Mapping
+
+from django.conf import settings
+from django.test import override_settings
+
+from sentry import options
+from sentry.conf.types.celery import SplitQueueSize
+from sentry.queue.routers import SplitQueueRouter
+from sentry.testutils.pytest.fixtures import django_db_all
+from sentry.utils.celery import make_split_queues
+
+
+@django_db_all
+def test_split_router_legacy() -> None:
+    queues = [
+        "post_process_transactions_1",
+        "post_process_transactions_2",
+        "post_process_transactions_3",
+    ]
+    queues_gen = itertools.cycle(queues)
+    with override_settings(
+        SENTRY_POST_PROCESS_QUEUE_SPLIT_ROUTER={
+            "post_process_transactions": lambda: next(queues_gen),
+        },
+        CELERY_SPLIT_QUEUE_ROUTES={},
+    ):
+        router = SplitQueueRouter()
+        assert router.route_for_queue("save_event") == "save_event"
+        assert router.route_for_queue("post_process_transactions") == "post_process_transactions_1"
+        assert router.route_for_queue("post_process_transactions") == "post_process_transactions_2"
+        assert router.route_for_queue("post_process_transactions") == "post_process_transactions_3"
+
+        legacy_mode = options.get("celery_split_queue_legacy_mode")
+        options.set("celery_split_queue_legacy_mode", [])
```

Comment:
You can use `override_options` to modify options values via a context manager similar to `override_settings`

Reply:
damn, that's much better, thanks.